### PR TITLE
cmake : use list(APPEND ...) instead of set() + dedup linker

### PR DESCRIPTION
**Pull request by ggerganov**
ref https://github.com/ggerganov/llama.cpp/pull/9339#issuecomment-2346950692

- Using `list(APPEND X Y)` is considered better practice compared to `set(X X Y)`.
- Add `list(REMOVE_DUPLICATES ${GGML_EXTRA_LIBS})` to deduplicate flags coming from Accelerate and Metal branches (and possibly other cases too)

---

- [x] I have read the [contributing guidelines](https://github.com/ggerganov/llama.cpp/blob/master/CONTRIBUTING.md)
- Self-reported review complexity:
  - [ ] Low
  - [ ] Medium
  - [ ] High

### CODE REVIEW

File: ggml/src/CMakeLists.txt [hunk 1]
```diff
@@ -504,7 +506,8 @@ if (GGML_HIPBLAS)
         message(FATAL_ERROR "Static linking not supported for HIP/ROCm")
     endif()
 
-    set(GGML_EXTRA_LIBS ${GGML_EXTRA_LIBS} PUBLIC hip::host roc::rocblas roc::hipblas)
+    # TODO: this "PUBLIC" here seems wrong
+    list(APPEND GGML_EXTRA_LIBS PUBLIC hip::host roc::rocblas roc::hipblas)
```
Comment (Green-Sky) [inline ggml/src/CMakeLists.txt:510]: I am surprised this hack works at all. But I guess its strings internally...
But that also means that everything following will be made public.

Comment (ggerganov) [reply ggml/src/CMakeLists.txt:510]: Yup me too. I don't have HIP environment setup to test and find ways to avoid it.

**Comment by Xarbirus** [comment]
@ggerganov , I added SYCL fix in #9469

**Comment by Xarbirus** [comment]
And this one #9471 please.

**Comment by ggerganov** [comment]
@Xarbirus Looks like we broke the Docker CI with this change:

https://github.com/ggerganov/llama.cpp/actions/runs/10860757616/job/30141751962

Do you have any suggestions for a fix?

Will tag also the Intel team for suggestions: @abhilash1910 @airMeng @NeoZhangJianyu @luoyu-intel

**Comment by Xarbirus** [comment]
I have only one suggestion: #9487 (`set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} -fsycl")`), PTAL

**Pull request merged**